ATOMIC_SWAP.md

```diff
@@ -0,0 +1,335 @@
+# Atomic Swap
+
+This is a how-to of using the atomic-swap feature of the wallet.
+Both P2PKH and P2SH (MultiSig) can participate on the atomic-swap.
+
+## Description
+
+An atomic-swap transaction is a transaction with multiple tokens and inputs from different wallets.
+The atomic-swap process is how participants can safely coordinate the content and signing of a transaction.
+
+## Atomic-swap process
+
+The process can be divided in 2 phases:
+1. Negotiation phase
+  - Participants will coordinate the content of the atomic-swap.
+1. Signing phase
+  - Participants will sign the proposed transaction and exchange the signatures.
+  - A participant will collect the signatures and push the signed transaction on the network.
+
+### Simulation
+
+Let's divide the steps from an atomic-swap using a simulated exchange between Alice and Bob.
+
+1. Alice and Bob agree that Bob will send 10 HTR to Alice in exchange for 20 TKa (Token Alice).
+2. Bob starts a proposal with his side of the swap (sending 10 HTR and receiving 20 TKa to one of his addresses).
+    - Uses `POST /wallet/atomic-swap/tx-proposal`
+    - He will send the serialized proposal to Alice.
+    - Alice will check the contents of the proposal with the decode api
+3. Alice will update the proposal with her side of the swap (sending 20 TKa and receiving 10 HTR to one of her addresses).
```

Comment:
Bob already said that he will receive 20 TKa, Alice needs to say the same to the proposal API? Which parameters should be used?